homeassistant/components/climate/__init__.py

```diff
@@ -334,6 +339,19 @@ def __getattribute__(self, __name: str) -> Any:
             "_ClimateEntity__mod_supported_features"
         )
 
+    def __init_subclass__(cls, **kwargs: Any) -> None:
+        """Post initialisation processing."""
+        super().__init_subclass__(**kwargs)
+        if (
+            "is_aux_heat" in cls.__dict__
+            or "_attr_is_aux_heat" in cls.__dict__
```

Comment:
This won't work if `_attr_is_aux_heat` is set dynamically.
I think we should add a check here instead: https://github.com/home-assistant/core/blob/c78e089526076a6ede9bed02fb67d17182e26d4f/homeassistant/components/climate/__init__.py#L509-L510